frontend/src/components/Common/Chart/index.tsx

```diff
@@ -153,16 +157,27 @@ const Chart = memo(
       );
     }, [colorShuffle, config.colors, nshades]);
 
-    const labels = useMemo(() => {
+    const [labels, fullDatesForCsv] = useMemo<
+      [(string | number)[], (string | number)[]]
+    >(() => {
       if (!transpose) {
-        return indices.map(index => header[index]);
+        const ret = indices.map(index => header[index]);
+        return [ret, ret];
       }
-      return tableRows.slice(chartRange[0], chartRange[1]).map(row => {
-        const dateFormat = data.EWSConfig ? 'HH:mm' : 'YYYY-MM-DD';
-        return moment(row[config.category])
-          .locale(t('date_locale') as LocaleSpecifier)
-          .format(dateFormat);
-      });
+      const fullLabels = tableRows
+        .slice(chartRange[0], chartRange[1])
+        .map(row => {
+          return moment(row[config.category])
+            .locale(t('date_locale') as LocaleSpecifier)
+            .format('YYYY-MM-DD HH:mm');
+        });
+
+      return data.EWSConfig
+        ? [fullLabels.map(x => x.split(' ')[1]), fullLabels]
+        : [
+            fullLabels.map(x => x.split(' ')[0]),
+            fullLabels.map(x => x.split(' ')[0]),
```

Comment:
You're returning the same values twice, is this normal? Should the second part be `fullLabels` instead for consistency with above?